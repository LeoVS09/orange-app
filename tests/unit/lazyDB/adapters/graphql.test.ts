import cons from "../../utils/console.mock"

jest.mock('graphql-tag');

const queryByIdVariables = { id: { type: 'UUID', value: 1, required: true } }

describe('Mapper scheme to graphql query', () => {

    // Should be inside for mock graphql-tag
    const { generateQuery } = require('@/lazyDB/adapters/graphql')

    beforeEach(() => {
        cons.mockConsole()
    })

    afterEach(() => {
        cons.restoreConsole()
    })

    it('should generate query for read list of entities', () => {
        const operation = 'countries'
        const fields = [{
            nodes: ["name", "code", "updatedAt", "id", "nodeId", ]
        }]

        const query = generateQuery({ operation, fields })

        expect(query).toMatchSnapshot()
    })


    it('should generate query for read list of entities without id', () => {
        const operation = 'countries'
        const fields = [{
            nodes: ["name", "code", "updatedAt"]
        }]

        const query = generateQuery({ operation, fields })

        expect(query).toMatchSnapshot()
    })

    it('should generate query for read list of entities without only nodeId', () => {
        const operation = 'countries'
        const fields = [{
            nodes: ["name", "id",  "code", "updatedAt"]
        }]

        const query = generateQuery({ operation, fields })

        expect(query).toMatchSnapshot()
    })

    it('should generate query for read list of entities without only id', () => {
        const operation = 'countries'
        const fields = [{
            nodes: ["name", "nodeId", "code", "updatedAt"]
        }]

        const query = generateQuery({ operation, fields })

        expect(query).toMatchSnapshot()
    })



    it('should generate query for read entity by id with one to many links', () => {
        const operation = 'country'
        const fields = [
            "createdAt",
            "updatedAt",
            "nodeId",
            "id",
            "name",
            {
                cities: ["id", "name","nodeId",  "updatedAt"]
            }
        ]

        const query = generateQuery({ operation, fields, variables: queryByIdVariables })

        expect(query).toMatchSnapshot()
    })

    it('should generate query for read entity by id with one to many links without parent id', () => {
        const operation = 'country'
        const fields = [
            "createdAt",
            "updatedAt",
            "name",
            {
                cities: ["id", "nodeId", "name", "updatedAt"]
            }
        ]

        const query = generateQuery({ operation, fields, variables: queryByIdVariables })

        expect(query).toMatchSnapshot()
    })

    it('should generate query for read entity by id with one to many links without child id', () => {
        const operation = 'country'
        const fields = [
            "id",
            "nodeId",
            "createdAt",
            "updatedAt",
            "name",
            {
                cities: ["name", "updatedAt"]
            }
        ]

        const query = generateQuery({ operation, fields, variables: queryByIdVariables })

        expect(query).toMatchSnapshot()
    })


    it('should generate query for read entity by id with one to one link', () => {
        const operation = 'city'
        const fields = [
            "id",
            "nodeId",
            "createdAt",
            "updatedAt",
            "name",
            {
                country: ["id", "nodeId", "name", "updatedAt"]
            }
        ]

        const query = generateQuery({ operation, fields, variables: queryByIdVariables })

        expect(query).toMatchSnapshot()
    })

    it('should generate query for read entity by id with one to one link wihout id', () => {
        const operation = 'city'
        const fields = [
            "createdAt",
            "updatedAt",
            "name",
            {
                country: ["name", "updatedAt"]
            }
        ]

        const query = generateQuery({ operation, fields, variables: queryByIdVariables })

        expect(query).toMatchSnapshot()
    })
})

