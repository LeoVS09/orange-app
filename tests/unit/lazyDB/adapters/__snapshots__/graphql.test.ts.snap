// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mapper scheme to graphql query should generate query for read entity by id with one to many links 1`] = `
"
query Country($id: UUID!) {
  country(id: $id) {
    id
    nodeId
    createdAt
    updatedAt
    name
    cities {
      id
      nodeId
      name
      updatedAt
    }
  }
}
"
`;

exports[`Mapper scheme to graphql query should generate query for read entity by id with one to many links without child id 1`] = `
"
query Country($id: UUID!) {
  country(id: $id) {
    id
    nodeId
    createdAt
    updatedAt
    name
    cities {
      name
      updatedAt
    }
  }
}
"
`;

exports[`Mapper scheme to graphql query should generate query for read entity by id with one to many links without parent id 1`] = `
"
query Country($id: UUID!) {
  country(id: $id) {
    createdAt
    updatedAt
    name
    cities {
      id
      nodeId
      name
      updatedAt
    }
  }
}
"
`;

exports[`Mapper scheme to graphql query should generate query for read entity by id with one to one link 1`] = `
"
query City($id: UUID!) {
  city(id: $id) {
    id
    nodeId
    createdAt
    updatedAt
    name
    country {
      id
      nodeId
      name
      updatedAt
    }
  }
}
"
`;

exports[`Mapper scheme to graphql query should generate query for read entity by id with one to one link wihout id 1`] = `
"
query City($id: UUID!) {
  city(id: $id) {
    createdAt
    updatedAt
    name
    country {
      name
      updatedAt
    }
  }
}
"
`;

exports[`Mapper scheme to graphql query should generate query for read list of entities 1`] = `
"
query Countries {
  countries {
    nodes {
      id
      nodeId
      name
      code
      updatedAt
    }
  }
}
"
`;

exports[`Mapper scheme to graphql query should generate query for read list of entities without id 1`] = `
"
query Countries {
  countries {
    nodes {
      name
      code
      updatedAt
    }
  }
}
"
`;

exports[`Mapper scheme to graphql query should generate query for read list of entities without only id 1`] = `
"
query Countries {
  countries {
    nodes {
      nodeId
      name
      code
      updatedAt
    }
  }
}
"
`;

exports[`Mapper scheme to graphql query should generate query for read list of entities without only nodeId 1`] = `
"
query Countries {
  countries {
    nodes {
      id
      name
      code
      updatedAt
    }
  }
}
"
`;
